tests/connectors/studio/test_studio.py

```diff
@@ -151,3 +173,28 @@ def test_submit_from_tracer_works_with_empty_tracer(
     empty_trace_id_list = studio_client.submit_from_tracer(tracer)
 
     assert len(empty_trace_id_list) == 0
+
+
+def test_can_upload_dataset_with_minimal_request_body(
+    studio_client: StudioClient,
+    examples: Sequence[Example[str, str]],
+) -> None:
+    dataset_repo = InMemoryDatasetRepository()
+    dataset = dataset_repo.create_dataset(examples, "my_dataset")
+    result = studio_client.submit_dataset(dataset=dataset, examples=examples)
+    assert result is not None
+
+
+def test_can_upload_dataset_with_complete_request_body(
+    studio_client: StudioClient,
+    examples: Sequence[Example[str, str]],
+    labels: set[str],
+    metadata: dict[str, Any],
+) -> None:
+    dataset_repo = InMemoryDatasetRepository()
+    dataset = dataset_repo.create_dataset(
+        examples, "my_dataset", labels=labels, metadata=metadata
+    )
+    result = studio_client.submit_dataset(dataset=dataset, examples=examples)
+
+    assert result is not None
```

Comment:
we could check that the dataset we get back has the metadata / labels etc that we expect

Reply:
Added an "has been called with" assertion type